### DISCUSSION TITLE
dataloader feature for regular Vite projects

### DISCUSSION THREAD

**Discussion by gureckis** [vuejs/vitepress#3824] [Q&A]
the dataloader feature of Vitepress is neat but I'm having trouble finding a similar plugin for Vite by itself.  it is possible this functionality could be moved to a general plugin for use in other projects?

**Comment by brc-dd** [comment]
This is the plugin - https://github.com/vuejs/vitepress/blob/main/src/node/plugins/staticDataPlugin.ts - I don't think it has any vitepress specific logic. You can probably just copy and save it locally and register it in your vite.config.ts

**Comment by gureckis** [comment]
For future reference @Shakeskeyboarde put together something similar here: https://www.npmjs.com/package/vite-plugin-data

**Comment by Shakeskeyboarde** [comment]
That has some jems in it I'm going to borrow though :). Thanks for the mention @gureckis.
